src/state/presentationManagerImpl.test.ts

```diff
@@ -334,4 +340,23 @@ describe('presentationManager', () => {
 
     expect(await states).toEqual([{ type: 'idle' }]);
   });
+
+  it('should toggle the edit mode', () => {
+    presentationManager.toggleEditMode();
+    presentationManager.toggleEditMode();
+
```

Comment:
Please also check if `observePresentationState` emits the expected state in this situation. See "should start the presentation" for an example.